src/go/k8s/api/redpanda/v1alpha2/interfaces.go

```diff
@@ -0,0 +1,47 @@
+package v1alpha2
+
+import "sigs.k8s.io/controller-runtime/pkg/client"
+
+// The interfaces below are leveraged for consistently
+// creating clients to the internal cluster APIs in
+// our internal client factory code. Any CRD for which
+// reconciliation requires configuring a cluster should
+// implement some subset of these interfaces.
```

Comment:
I had contemplated introducing these interfaces in the `client` subpackage, but figured due to the interface for the factory supporting passing generic `client.Object`s, it was worth putting them here to do some type assertions on our implementing CRDs. That wouldn't have been possible putting the interfaces there though due to circular dependencies and I wanted to avoid doing some other subpackage just for interfaces.